pulumi/infra/api_gateway.py

```diff
@@ -0,0 +1,112 @@
+import json
+from typing import List, Optional
+
+import pulumi_aws as aws
+
+import pulumi
+
+API_GATEWAY_LOGS_JSON_FORMAT = json.dumps(
+    {
+        # This is verbatim what you get from clicking the JSON button at
+        # https://console.aws.amazon.com/apigateway/main/monitor/logging/edit
+        "requestId": "$context.requestId",
+        "ip": "$context.identity.sourceIp",
+        "requestTime": "$context.requestTime",
+        "httpMethod": "$context.httpMethod",
+        "routeKey": "$context.routeKey",
+        "status": "$context.status",
+        "protocol": "$context.protocol",
+        "responseLength": "$context.responseLength",
+        # And some extra- integration-specific ones.
+        "integrationError": "$context.integration.error",
+        "integrationStatus": "$context.integration.status",
+        "responseType": "$context.error.responseType",
+        "message": "$context.error.message",
+    }
+)
+
+
+class ApiGateway(pulumi.ComponentResource):
+    def __init__(
+        self,
+        name: str,
+        nomad_agents_alb_security_group: pulumi.Input[str],
+        nomad_agents_alb_listener_arn: pulumi.Input[str],
+        private_subnet_ids: pulumi.Input[List[str]],
```

Comment:
It's probably worth documenting what each of these inputs is for, and the relationship between them. For instance, what do `private_subnet_ids` do? How (if at all) are they related to the security group and listener ARN?

Reply:
i've more-accurately-named and documented these three.